### ISSUE TITLE
stable-diffusion-ui is always taking 3.4G of dedicated GPU memory even if it doesn't do anything

### ISSUE THREAD

**Issue by van1027** [easydiffusion/easydiffusion#1306]
**Describe the bug**
I am running stable-diffusion-ui and stable-diffusion-webui in parallel and noticed that when stable-diffusion-ui is running (even if it doesn't do anything), stable-diffusion-webui is often failing with OutOfMemory error. I started to look for solution but nothing worked. Then I found out that stable-diffusion-ui is always taking GPU memory. See the screenshot. It's taken right after the stable-diffusion-ui started up and nobody used it.

**To Reproduce**
1. Start stable-diffusion-ui
2. Open task manager
3. Go to Details tab
4. Right click on the columns
5. Click Select columns
6. Select Dedicated GPU memory and Command line
7. Sort by Dedicated GPU memory descending

**Expected behavior**
It should not hold on GPU memory if it doesn't do anything with it. Or there should be an option to disable that behaviour and unload dedicated GPU memory as soon as the last task is done.

**Screenshots**
![screenshot](https://github.com/cmdr2/stable-diffusion-ui/assets/131382639/8294b821-e54d-412d-8760-6f6c0e8ff17c)

**Desktop (please complete the following information):**
 - OS: Windows 10

**Comment by lucavice** [comment]
I'm not part of the dev team, but I think this is by design.

Immediately after you boot the web-ui, the GPU usage is actually zero. A few seconds after boot, you'll see in the command line that the tool will load the most recently used model:
![image](https://github.com/cmdr2/stable-diffusion-ui/assets/14258744/3fece134-b26d-4fe7-8464-a21630cac17b)

This means it will be pre-lodead in memory and ready for use for the first prompt.
I think this makes sense, as in 95% of cases, that's what most users would do and it saves time instead waiting for loading AFTER you asked to generate an image.

> It should not hold on GPU memory if it doesn't do anything with it

I do not agree with this, since loading models in GPU memory takes time (in my case, about 5-6 seconds), and you definitely don't want to wait the unload time and load time every time you want to make a little change on the prompt while using the same model.

**Comment by JeLuF** [comment]
If it would only be 5-6 seconds to load a model, unloading would be an option. For many, it takes a minute.

**Comment by arrowgent** [comment]
it makes sense to leave the model loaded in ram if youre using the application

if you want, a button to unload the model and then modifications for each time you press "make image" to check if it is loaded and if not then load the models.

basically unloading models would be pausing the application.  since maybe you want to use the computer for something else then come back to making images later.

**Comment by van1027** [comment]
> > basically unloading models would be pausing the application. since maybe you want to use the computer for something else then come back to making images later.
> 
> You could stop the application instead, and start it again when you're done with your other tasks.

If you are using your computer as a server and sharing with other people, you cannot stop anything for doing something else. Instead each app has to be memory efficient. If no one is using easy diffusion for several minutes it should release all the memory it took for caching, so other apps can use it.  That's my opinion.

**Comment by jlfranklin** [comment]
Perhaps unloading the model after a certain amount of idle time, say 5 minutes by default, but configurable.
